backend/src/main/java/com/bakdata/conquery/commands/PreprocessorCommand.java

```diff
@@ -100,57 +103,78 @@ protected void run(Environment environment, Namespace namespace, ConqueryConfig
 			pool = Executors.newFixedThreadPool(config.getPreprocessor().getNThreads());
 		}
 
-		final Collection<TableImportDescriptor> descriptors = new ArrayList<>();
-
 		// Tag if present is appended to input-file csvs, output-file cqpp and used as id of cqpps
 
-		isFailFast = namespace.get("fast-fail") == null ? false : namespace.get("fast-fail");
+		isFailFast = namespace.get("fast-fail") != null && namespace.getBoolean("fast-fail");
+		isStrict  = namespace.get("disable-strict") == null || !namespace.getBoolean("disable-strict");
 
-		final List<String> tags = namespace.getList("tag") != null ? namespace.getList("tag") : Collections.singletonList(null);
+		final List<String> tags = namespace.getList("tag");
 
-		if (namespace.get("in") != null && namespace.get("desc") != null && namespace.get("out") != null) {
-			log.info("Preprocessing from command line config.");
+		final File inDir = namespace.get("in");
+		final File outDir = namespace.get("out");
+		final List<File> descriptionFiles = namespace.<File>getList("desc");
 
-			for (String tag : tags) {
-				descriptors.addAll(findPreprocessingDescriptions(environment.getValidator(), new PreprocessingDirectories[]{
-						new PreprocessingDirectories(namespace.get("in"), namespace.get("desc"), namespace.get("out"))
-				}, tag));
-			}
 
+
+		log.info("Preprocessing from command line config.");
+
+		final Collection<PreprocessingJob> jobs = new ArrayList<>();
+
+		if (tags == null || tags.isEmpty()) {
+			for (File desc : descriptionFiles) {
+				final List<PreprocessingJob> descriptions =
+						findPreprocessingDescriptions(desc, inDir, outDir, Optional.empty(), environment.getValidator());
+				jobs.addAll(descriptions);
+			}
 		}
 		else {
 			for (String tag : tags) {
-				log.info("Preprocessing from config.json");
-				descriptors.addAll(findPreprocessingDescriptions(environment.getValidator(), config.getPreprocessor().getDirectories(), tag));
+				for (File desc : descriptionFiles) {
+					final List<PreprocessingJob> jobDescriptions =
+							findPreprocessingDescriptions(desc, inDir, outDir, Optional.of(tag), environment.getValidator());
+
+					jobs.addAll(jobDescriptions);
+				}
 			}
 		}
 
+		// This will halt preprocessing immediately.
+		if(isStrict) {
+			for (PreprocessingJob job : jobs) {
+				for(TableInputDescriptor input : job.getDescriptor().getInputs()) {
+					final File sourceFile = Preprocessor.resolveSourceFile(input.getSourceFile(), job.getCsvDirectory(), job.getTag());
+					if(!sourceFile.exists()) {
+						log.warn("Did not find file `{}` for preprocessing.", sourceFile);
+						addMissing(job);
+					}
+				}
+			}
+		}
 
-		descriptors.removeIf(Predicate.not(Preprocessor::requiresProcessing));
+		jobs.removeIf(Predicate.not(Preprocessor::requiresProcessing));

```

Comment:
Preprocessor::requiresProcessing wird nur in dieser Datei genutzt oder? Dann kann sie auch hier her